### PR TITLE
So basically bodybags of holding are just really deep bodybags

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Bodybags of holding now no longer fold up with people or objects in them.

They also use plastic and less plasma to make. Iunno, I feel like more stuff should use plastic where it makes sense. It's even in the same department, so the medical doctors can just go make some easy peezy.

## Why It's Good For The Game

Back when I made that BoH change I was talking a lot with people, namely some maintainers, about how basically these were not only a much bigger issue than bag storage, being able to haul even stuff that normally cannot be picked up (such as mech parts and literally people in a GBJ they have to spend an absurd amount of time to escape from), but were able to do some truly fucking disgusting things. I'm arbitrarily cashing in the thumbs up I got I think either from skoglol or cobby to pr this way back when.

Let's get a quick rundown on what a bodybag of holding can do currently:

A) You could store toxins bombs into boxes using these. This lets you maxcap people with boxes, and hide maxcaps in plain sight.
B) You could abduct people in the blink of an eye and there is seemingly no way out in a reasonable time before you're deleted using various techniques. Mostly, throwing the bag into space or into the cremator. It's a pretty easy murderbone with it too since you need only shove someone for a few seconds to do it.
C) You could put these in storage implants to haul massive amounts of machinery with you anywhere. This makes storage implants capable of some truly absurd things, like carrying an entire science department worth of stuff in a hidden compartment, or the aforementioned mechs. Nobody expects someone to make a durand in permabrig. (admittedly this is pretty dope but alas)
D) You could vore entire departments with them by putting someone into one and then putting the folded bag into your chest cavity.
E) They weren't being used as bodybags at all.

Basically, rather than completely deleting them, I just removed the problem part of them. They work like bodybags now.

As for why they use plastic. Uh. I think more things should be made from plastic. Yep yep.

## Changelog
:cl:
balance: Bodybags of holding now no longer fold up with objects inside them.
balance: Bodybags use plastic rather than iron in their construction.
/:cl:

### CODE REVIEW

**Comment by EdgeLordExe** [comment]
just make it so zipping someone while he is alive takes like 5 seconds.

**Comment by necromanceranne** [comment]
The abduction bit is only one of many concerns. Think less about the abduction, and think more about how little this ever gets used for their actual purpose because the meme use happens to be a powergay wetdream.

**Comment by trerri** [comment]
What if you could stick in a bluespace anomaly core to retain the old effect?
What if you could store one person in, and they could immediately get out if they had control of their mob?
What if they had an alert on their sprite, like the morgue trays'?
Hate to sound like a pl*yer, but improve, don't remove. (or ideally improve while removing as little cool things as possible)

**Comment by Barhandar** [comment]
This PR makes it into a useless overpriced version of the regular bodybag, because if you have multiple corpses, you can either grab a box of bodybags or don't have time to get them all to medbay all bodybagged.

Is it really that hard to solve the vacuum/abduction problem by making getting out instant, adding maximums of size and quantity to the items that can be carried within, and making it not fit into storage implant (moreso since IIRC it's fluffed as bluespace anyway, so you can make it conflict like how BoH interferes with teleports).

**Comment by anconfuzedrock** [comment]
if you compromise at all, make it so folded bluespace bags with stuff in them are just giant items so you can use them to carry bodies easier but only in your hands. even then bluespace bags would be crazy good so I'm not saying do that, just do if  you decide to reduce the nerfness of this nerf at all.

**Comment by necromanceranne** [comment]
I'm pretty happy to reduce the material cost even further. It's fairly minuscule actually already.

**Comment by Barhandar** [comment]
Just remove bodybags of holding entirely if you want to keep on with "cannot be folded with a corpse".

Speaking of which, how about making it unable to fold with items or alive entities? I.e. can only fold up corpses and whatever the corpse has on, and too large to storageimplant. Ought to be much harder to vacuum anything that way.

**Comment by necromanceranne** [comment]
I feel like this is the most sensible change. I don't think outright deleting them is really necessary, they can still fill a limited niche as an object hauler by dragging it along if necessary.

**Comment by Tharcoonvagh** [comment]
What differentiates bs bags from normal bags then

**Comment by Barhandar** [comment]
Being able to drag entire department's worth of items doesn't really solve the problem of stealing a department's worth of items.

And again: with this change it's no longer worth it _in_ the intended capacity. Just use a regular bodybag.

**Comment by zxaber** [comment]
Sorta removes any fun that can be had with it. And to be honest, I'd say most of the issues could be solved by:

1:Remove any delay in exiting the bag; it's useful for transporting willing people but easy to escape
2:Make explosion sources within a bluespace spacial distortion object just damage/destroy the object and not actually create an explosion

People rarely use normal bodybags as it is, you might as well just remove the item if you're making these changes.

**Comment by ATH1909** [comment]
> if you compromise at all, make it so folded bluespace bags with stuff in them are just giant items so you can use them to carry bodies easier but only in your hands. even then bluespace bags would be crazy good so I'm not saying do that, just do if you decide to reduce the nerfness of this nerf at all.

Making them bulky items when folded sounds like a good idea.

I don't think that escaping them should be made instant, as abducting people with bluespace body bags is, like, 75% the fun of the item now.

If necessary, making bluespace body bags require an anomaly core alongside these other changes should also work.

**Comment by ATH1909** [comment]
I've made a PR to make bluespace body bags gigantic items if they're closed while they have something in them, as well as some other stuff (including making sure that you can't put them inside of bags of holding).

I'll test it and then post it in a few hours.

**Comment by Doosteroni** [comment]
Hi, Fixed your PR title for you:
makes bodybags of holding hot garbage that nobody will ever use

**Comment by Megarop** [comment]
> As for why they use plastic. Uh. I think more things should be made from plastic. Yep yep.

Why do you want more things to be made from plastic if there's so little reason for more things to be made from plastic that we have to resort to making random items require it?
 
> The abduction bit is only one of many concerns. Think less about the abduction, and think more about how little this ever gets used for their actual purpose because the meme use happens to be a powergay wetdream.

The bluespace bodybag rarely gets used for its "actual purpose" beacuse its actual purpose isn't that useful. Getting rid of its other uses isn't gonna suddenly make its remaining functions any less useless.
Besides, it's ss13, being able to use items or game mechanics for something other than their intended purpose is par for the course and part of the reason people like the game.

**Comment by cacogen** [comment]
Instead of deleting the function to contain people you could've just made the bags easier to escape from and harder to trap people in. The other things are legitimate issues but you undermine this as a fix for them by needlessly removing that mechanic. If you want a better received pull request you'd retain that function.

**Comment by XDTM** [comment]
I believe that fitting large entities into inventory storage is a mechanic that adds a lot to gameplay and creativity, and should stay in some form, even if not necessarily a bodybag.

A) As others have said, the bomb issue could be solved by making the bag absorb the explosion and break insted of placing it outside.
B) Trapping people would be solved by adding a small delay in closing the bag around mobs/large objects. (I'd avoid making escape instant, since trapping people in your pocket is, in my opinion, a pretty fun mechanic. This way you could still abduct people if you restrain them first)
C) Can't see the problem here, if you have a cavity implant, the resources to make a bluespace bodybag and a full set of mech parts, time and foresight enough to do so, and don't get caught while assembling the mech, you deserve the reward and it makes for a fun story.
D) If this is an actual problem it's solvable by putting limits of the carry capacity of the bag.
E) The functionality could be moved onto a specialized gadget instead of a bodybag, which i would welcome since it makes it more intuitive to non-experienced players, but would require a new sprite and some code adjustments.

**Comment by anconfuzedrock** [comment]
Bluespace body bags were ludicrously overpowered beforehand, you can't be surprised that the nerf is this severe.

**Comment by Winterous** [comment]
First of all: It is absolutely hilarious to put a male and female corgi in a folded bodybag, and implant it in someone's chest.
Baby corgis just pop into existence at their feet, it's a horrifying thing that I would hate to see removed.

Second, I think that adding a like, 3-5 second delay to FOLD UP the bag if it contains anything would be a good solution. In addition to that, making things not explode / smoke / foam / whatever when inside a folded bag seems like a good idea, but there's gonna be a LOT of individual cases to consider.
Like gold slime core reaction, life-making chemical reaction, flashbangs, barrier grenades, quantum-spin-inverters, etc.. What things should function in that pocket dimension, what things should not? Why should or should not they function?

**Comment by cacogen** [comment]
It would be preferable to players to either blacklist troublesome items or to somehow limit their effects to the contents of the pocket dimension. I don't know how feasible it would be but you could make the pocket dimension a physical location the contents of which is moved to the bag when opened. This way anything that happens in the bag stays in the bag and properly affects other things in the bag.

**Comment by ExcessiveUseOfCobblestone** [comment]
https://github.com/tgstation/tgstation/pull/53760#issuecomment-693476906

**Comment by necromanceranne** [comment]
Given ATHATH has already indicated he has no intent on changing their kidnapping power, just their ability to be stored in bags, I don't if his change will address that at all or he will make any changes in that regard. I genuinely don't care to implement any half measures. I honestly think they should just be a step up from body bags or even lockers for loose object storage and that's it really.

**Comment by TreyWJ** [comment]
This PR hurts my soul. Bluespace body bags were one of my favorite items just because of how funny they were. I remember being in a round during a plasmaflood, and a chaplain used one to bring me and like 4-5 other people around the station. Then he released us on a xeno nest as a little mini-army.

The point is that they're _fun_. Who cares if people aren't using them solely to drag more dead bodies? All of your concerns are easily addressed by changing them, not just removing the feature.

A) Prevent folded bluespace bodybags from fitting in boxes.
B) I do not see the huge problem with "abductions" you seem to be hellbent on "fixing." To abduct someone in it, you need to first unfold the bag onto the ground, have them stand on it, zip it up, and grab it before they can simply walk right out. But even if this were a problem, **there is a much better fix!** Just make it so they have to be laying down to be zipped into the bag. And add 2 seconds of time to "zip" the bag. Now abducting someone becomes much more tedious than just stunning and cuffing (which allows you to cremate and roundend people just the same).
C) Make them unable to fit in storage implants.
D) That sounds funny and would need to be consensual to actually work. Again these are fun mechanics that make for happy times, I have literally never seen this done maliciously because if you want to be malicious there are better ways than getting 10 people into a fuckin bluespace bodybag. 
E) Who cares??? It's fun!

**Comment by ExcessiveUseOfCobblestone** [comment]
If he wont properly address it itll get closed and then ill look into this PR a bit more.

**Comment by TreyWJ** [comment]
> If he wont properly address it itll get closed and then ill look into this PR a bit more.

He already mentioned ideas to make it easier to escape and so you'd have to be resting to get put in it. It was also suggested that a delay for zipping be added. At that point kidnapping is no easier than stun and cuffing. Why would that be insufficient?

**Comment by necromanceranne** [comment]
he could totes do that in his pr if he wants. Nothing stopping him and it'd address Cobby's desires to resolve their kidnapping power.

**Comment by ExcessiveUseOfCobblestone** [comment]
> He already mentioned ideas to make it easier to escape and so you'd have to be resting to get put in it. It was also suggested that a delay for zipping be added. At that point kidnapping is no easier than stun and cuffing. Why would that be insufficient?

You sorta answered your own question, it wouldnt but again what he mentioned and what his ideas are currently aren't reflected in the PR except as comments.

**Comment by JenqaDev** [comment]
you just want to ruin our fun

**Comment by TheSilverNuke** [comment]
Anne please stop removing all the bluespace content. Bags of Holdings are all but gone at this point, please don't take away the bodybags too. Less content for the station is only going to harm us.

**Comment by necromanceranne** [comment]
> you just want to ruin our fun
> EDIT: because of the recent threats i have been given from unreliable sources i have to give me trustworthy(not) opinion(who cares):

Did you write this with a gun to your head? You okay there?

> Anne please stop removing all the bluespace content. Bags of Holdings are all but gone at this point, please don't take away the bodybags too. Less content for the station is only going to harm us.
> 
> Bluespace bodybags are frequently used by Paramedics to store multiple bodies before bringing them back to med. This is a great use for those moments when a murderboner goes on a killing spree. Faster retrieval means faster revival.
> 
> They also serve as massive storage if necessary, helping you to move a large quantity of objects from one place to another. Bags of Holding used to serve this purpose but now that you need a bluespace core for them I have not seen them in use literally once since you added that "feature."

You should probably look into toxins anomaly creation.

Also this doesn't change their storage capacity and in fact is still entirely usable by parameds for body retrival. You drag bodybags at full speed.

**Comment by JenqaDev** [comment]
p-please don't. if they see this I'm doomed...

**Comment by PKPenguin321** [comment]
it's ok to have overpowered things if theyre cool like this
